### PR TITLE
Configuration exception handling

### PR DESCRIPTION
**Pull request by chesterxgchen**
Fixes # .  Both customer reported  issue and FLARE-516 
### Description
**Problem**
When user use embedded component in configuration, i.e. component's argument is another component or class. such as
```
   "comp1": { 
      "path": "module1.for.comp1.Compt1", 
       "args" : {
            "path": "module2.for.comp2.Comp2", 
             args: {
                "a": "b"
             }
        }
   }

```
Where comp2 is the embedded component. The original component_builder looks the dictionary argument 
```
"args" : {
            "path": "module2.for.comp2.Comp2", 
             args: {
                "a": "b"
             }
        }
```
Couldn't sure this is normal dictionary arg or its embedded component configuration. So the Component_builder just try to build the component ( instantiate the class for Comp2). If the class Comp2 is instantiated, then ok. if failed, the component_builder is considered it is normal dictionary arg, so swallow the exception. 

This logic is flawed as it did not consider the cases where 
   * The configuration error for Comp2 arguments causing Comp2 class instantiation failure
   * The initialization error for Comp2 causing Comp2 class instantiation failure

as result, the user get very strange configuration error message or random exception which is not related the root cause of the exception. 

**Approaches**
This PR attempt to fix this with following approaches: 
* Try to identity the embedded configuration is a component/class before try to instantiate 
    * if the instantiation failed, we raise the exception for user ( this will cover both args mis-config and initiation error) 
    * The class/component identification can be done by checking 
               * must has a "path" or "name"
                * must have a valid class path
  * for non-class config, they must be normal dict

**Changes**
1) change the component_builder with above logics
2) add unit tests to cover both positive and negative of component (with/without embedded component) creations 
    Unit tests handle both < 3.9.x and 3.10.x python message differences. 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by yanchengnv — APPROVED** [review]
Yeah, I have been thinking about similar approach.

**Comment by yanchengnv** [comment]
There is a unit test error. Please fix.

**Review by holgerroth — APPROVED** [review]
Nice. It works as expected on my side.

**Comment by IsaacYangSLA** [comment]
One question.  In the resources.json.default file, one component does not have args.  Because args is required for component, will this PR think it's not a component?

**Comment by IsaacYangSLA** [comment]
Inside resources.json.default of server startup kit local folder
`        {
            "id": "job_store",
            "path": "nvflare.app_common.storages.filesystem_storage.FilesystemStorage"
        }
`

File: examples/advanced/cifar10/cifar10-sim/jobs/cifar10_central/cifar10_central/config/config_fed_server.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
       "args": {
         "model": {
           "path": "pt.networks.cifar10_nets.ModerateCNN",
-	      "args": {}
+	      "args": {
+          }
```
Comment (YuanTingHsieh) [inline examples/advanced/cifar10/cifar10-sim/jobs/cifar10_central/cifar10_central/config/config_fed_server.json:20]: can we undo changes to this file? thanks

File: nvflare/fuel/utils/component_builder.py [hunk 2]
```diff
@@ -28,6 +27,20 @@ def get_module_scanner(self):
         """
         pass
 
+    def is_class_config(self, config_dict: dict) -> bool:
+        # the dict has both path/name or args, it's likely a component,
+        # test if it has a valid class path
+        if "args" in config_dict and ("path" in config_dict or "name" in config_dict):
```
Comment (YuanTingHsieh) [inline nvflare/fuel/utils/component_builder.py:33]: the comment string should be "the dict has both args and path/name"

What if the SUB component does not specify args?

So maybe somewhere in documentation we need to say you need to specify "args" for it to be treat as a SUB component?

**Review by YuanTingHsieh — APPROVED** [review]
